ansible/roles/kubernetes/tasks/debian.yml

```diff
@@ -0,0 +1,47 @@
+---
+# Instructions: https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#installing-kubeadm-kubelet-and-kubectl
+- name: add apt signing key for kubernetes (1/6)
+  apt_key:
+    url: https://packages.cloud.google.com/apt/doc/apt-key.gpg
+    state: present
+
+# Removed the following repo due to stability issues: https://apt.kubernetes.io/
+- name: adding apt repository for kubernetes (2/6)
+  apt_repository:
+    repo: deb http://packages.cloud.google.com/apt/ kubernetes-xenial main
+    state: present
+  register: kubernetes_repository
+  retries: 10
+  until: kubernetes_repository is success
+
+# To downgrade these packages use the following apt options:
+#   dpkg_options: force-downgrade
+#   force: yes
+- name: install kubernetes packages (3/6)
+  apt:
+    name: '{{ packages }}'
+    state: present
+    update_cache: yes
+  register: apt_install_kube
+  retries: 5
+  until: apt_install_kube is success
+  vars:
+    packages:
+      - kubelet={{ kubelet_version }}
+      - kubeadm={{ kubeadm_version }}
+      - kubectl={{ kubectl_version }}
+
+- name: add apt-mark hold to avoid version skew issues. (4/6)
```

Comment:
The 3 tasks below can be put into 1 task with a `loop` added to it. Less is more :)

Reply:
Combined, might need to check that it will still work with how I did it?